107-tee-boot-attestation.md

```diff
@@ -0,0 +1,498 @@
+encrypted VM image sealed with initial vTPM state, which is in-turn encrypted
+with a key stored within Keylime. This step involves a new Keylime Registrar
+endpoint to fetch the public components of its "TEE Attestation Key" to then
+encrypt the initial vTPM state.
+
+- User creates an encrypted disk image instance from a plain text image
+  template.
+- User creates initial vTPM state for the CVM.
+- User seals encrypted disk image key against the initial vTPM state.
+- User makes POST request to new Keylime Registrar endpoint to fetch TEE
+  attestation key. Keylime Registrar replies with public components of TEE
+  attestation key.
+- User encrypts initial vTPM state with public TEE Attestation Key.
+- User stores encrypted initial vTPM state along with encrypted root disk.
+
+A diagram showing the VM provisioning process:
+```
+                                      ┌────────────────┐
+                                      │                │
+                                      │ ┌────────────┐ │
+                                      │ │VM rootfs   │ │
+                                      │ │Sealed w/   │ │
+┌──────┐                              │ │vTPM State  │ │
+│ VM   │    ┌──────────────┐          │ │Key         │ │
+│ Disk ├───►│ Provisioning ├─────────►│ └────────────┘ │
+│ Image│    │     Tool     │          │                │
+└──────┘    └───┬─────▲────┘          │ ┌────────────┐ │
+                │     │               │ │Initial     │ │
+         GET    │     │    TEE        │ │vTPM State  │ │
+         TEE    │     │Attestation    │ │encrypted w/│ │
+     Attestation│     │Public Key     │ │TEE         │ │
+     Public Key │     │               │ │Attestation │ │
+              ┌─▼─────┴─┐             │ │Key         │ │
+              │ Keylime │             │ └────────────┘ │
+              │Registrar│             │   Encrypted    │
+              └─────────┘             │     Image      │
+                                      └────────────────┘
+```
+2. Pre-boot VM TEE Attestation
+
+With the image created and the CVM now running, SVSM is able to attest the CVM's
+TEE evidence, unlock its initial vTPM state, and decrypt its root disk.
+This step involves another new Registrar webhook to perform attestation on the
+TEE evidence.
+
+- Since SVSM relies on the untrusted host for networking, all incoming secrets
+  sent upon a successful attestation must be encrypted with a key that only the
+  CVM controls. Therefore, SVSM generates a "CVM Integrity Key" that Keylime
+  must use to encrypt all secrets sent back to SVSM.
+- SVSM hashes Agent UUID, CVM Integrity public key, and ID_BLOCK (or TDX/CCA/etc
+  equivalent) and requests an attestation report with this hash as REPORT_DATA.
+  This ensures freshness and integrity of attestation report.
+- SVSM POSTs UUID, CVM Integrity public key, attestation report, and encrypted
+  initial vTPM state to the Keylime Registrar.
```

Comment:
SVSM shouldn't be sending the actual vTPM state to the attestation server, only the "Encrypted vTPM state key".
ie the data handled by the attestation server needs to be exclusively read-only data. the vTPM state is read-write data, so it has to be the "Encrypted vTPM state key" that is unlocked by the attestation server